core/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -1031,13 +982,48 @@ synchronized void clear() {
                 residualMembers.clear();
         }
 
+        /**
+         * Spec: RTP17g
+         */
+        synchronized void reEnter() {
+            for (Map.Entry<String, PresenceMessage> entry: members.entrySet()) {
+                PresenceMessage member = entry.getValue();
+                member.action = PresenceMessage.Action.enter;
+                try {
+                    updatePresence(member, null);
```

Comment:
1. Could you elaborate on why you replaced
```
PresenceMessage itemToSend = new PresenceMessage();
itemToSend.clientId = item.clientId;
itemToSend.data = item.data;
itemToSend.action = PresenceMessage.Action.enter;
```
with
```
PresenceMessage member = entry.getValue();
member.action = PresenceMessage.Action.enter;
```
It's not obvious that that change is correct (won't it have a possibly-stale connectionId?)

2. Why did you remove the completionListener that was previously being passed to updatePresence? Is there some magic that will turn an async failure into an exception if a completionListener isn't passed that I'm missing?

Reply:
I guess I misinterpreted RTP17g.
Is this a good implementation https://github.com/ably/ably-java/pull/842/commits/28531da66223c2beed1753ea7df9e1729eb87d4e?